File: services/go-hex-uds/domain/entity/result.go
// Package entity holds the value objects of the UDS domain.
package entity

type ResultCode string

const (
	CodeOK           ResultCode = "OK"
	CodeBadRequest   ResultCode = "ERR_BAD_REQUEST"
	CodeInvalidToken ResultCode = "ERR_INVALID_TOKEN"
)

type VerificationResult struct {
	Valid   bool
	Subject string
	Code    ResultCode
}

File: services/rust-hex-uds/src/domain/result.rs
#[derive(Debug, Clone, PartialEq)]
pub enum DomainCode { Ok, BadRequest, InvalidToken }

impl DomainCode {
    pub fn as_str(&self) -> &'static str {
        match self { Self::Ok => "OK", Self::BadRequest => "ERR_BAD_REQUEST", Self::InvalidToken => "ERR_INVALID_TOKEN" }
    }
}

#[derive(Debug, Clone)]
pub struct VerificationResult { pub valid: bool, pub subject: String, pub code: DomainCode }
